src/openrct2/ride/Ride.cpp

```diff
@@ -6208,50 +6207,48 @@ void ride_get_entrance_or_exit_position_from_screen_position(
 
     ride = get_ride(gRideEntranceExitPlaceRideIndex);
     if (ride == nullptr)
-        return;
+        return entranceExitCoords;
 
     stationHeight = ride->stations[gRideEntranceExitPlaceStationIndex].Height;
 
-    screen_get_map_xy_with_z(screenX, screenY, stationHeight * 8, &mapX, &mapY);
+    screen_get_map_xy_with_z(screenCoords.x, screenCoords.y, stationHeight * 8, &mapX, &mapY);
     if (mapX == LOCATION_NULL)
     {
-        *outX = 0x8000;
-        return;
+        entranceExitCoords.x = LOCATION_NULL;
+        return entranceExitCoords;
     }
 
     word_F4418C = mapX;
     word_F4418E = mapY;
-    _unkF44188.x = floor2(mapX, 32);
-    _unkF44188.y = floor2(mapY, 32);
-    *outX = _unkF44188.x;
-    *outY = _unkF44188.y;
+
+    entranceExitCoords = { floor2(mapX, 32), floor2(mapY, 32), entranceExitCoords.z, INVALID_DIRECTION };
 
     if (ride->type == RIDE_TYPE_NULL)
-        return;
+        return entranceExitCoords;
```

Comment:
For these returns that look like failures I think it should really be setting .x to `LOCATION_NULL`. Not 100% sure on that. Need to investigate it further.

Reply:
I can set them to `LOCATION_NULL`, no hassle.